plugins/modules/idrac_secure_boot.py

```diff
@@ -0,0 +1,387 @@
+author:
+    - "Abhishek Sinha(@ABHISHEK-SINHA10)"
+notes:
+    - When this module runs in check_mode for I(import_certificates), it reports the changes found.
+    - This module does not support idempotency when I(import_certificates) is provided.
+    - This module supports IPv4 and IPv6 addresses.
+    - This module supports C(check_mode).
+"""
+
+EXAMPLES = """
+---
+- name: Import multiple SecureBoot certificate without applying to iDRAC.
+  dellemc.openmanage.idrac_secure_boot:
+    idrac_ip: "192.168.1.2"
+    idrac_user: "user"
+    idrac_password: "password"
+    ca_path: "/path/to/ca_cert.pem"
+    import_certificates: true
+    platform_key: /user/name/certificates/pk.pem
+    KEK:
+      - /user/name/certificates/kek1.pem
+      - /user/name/certificates/kek2.pem
+    database:
+      - /user/name/certificates/db1.pem
+      - /user/name/certificates/db2.pem
+    disallow_database:
+      - /user/name/certificates/dbx1.pem
+      - /user/name/certificates/dbx2.pem
+
+- name: Import a SecureBoot certificate and restart the server to apply it.
+  dellemc.openmanage.idrac_secure_boot:
+    idrac_ip: "192.168.1.2"
+    idrac_user: "user"
+    idrac_password: "password"
+    ca_path: "/path/to/ca_cert.pem"
+    import_certificates: true
+    platform_key: /user/name/certificates/pk.pem
+    restart: true
+    job_wait_timeout: 600
+"""
+
+RETURN = r'''
+---
+msg:
+  description: Status of the secure boot operation.
+  returned: always
+  type: str
+  sample: "Successfully imported the SecureBoot certificate."
+error_info:
+  description: Details of the HTTP Error.
+  returned: on HTTP error
+  type: dict
+  sample: {
+    "error": {
+      "code": "Base.1.0.GeneralError",
+      "message": "A general error has occurred. See ExtendedInfo for more information.",
+      "@Message.ExtendedInfo": [
+        {
+          "MessageId": "GEN1234",
+          "RelatedProperties": [],
+          "Message": "Unable to process the request because an error occurred.",
+          "MessageArgs": [],
+          "Severity": "Critical",
+          "Resolution": "Retry the operation. If the issue persists, contact your system administrator."
+        }
+      ]
+    }
+  }
+'''
+
+import json
+import os
+from urllib.error import HTTPError, URLError
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, IdracAnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_dynamic_uri, remove_key, validate_and_get_first_resource_id_uri,
+    trigger_restart_operation, wait_for_lc_status, get_lc_log_or_current_log_time)
+
+SYSTEMS_URI = "/redfish/v1/Systems"
+TIMEOUT_NEGATIVE_OR_ZERO_MSG = "The value for the 'job_wait_timeout' parameter cannot be negative or zero."
+SUCCESS_MSG = "Successfully imported the SecureBoot certificate."
+NO_OPERATION_SKIP = "Task is skipped as import is 'false'."
```

Comment:
NO_OPERATION_SKIP = "(Operation name) operation is skipped because the value provided for the parameter import_certificate is false."
		PROVIDE_ABSOLUTE_PATH = "Enter the absolute path of the (which certificate) certificate file." ->(please mention the certificate name)
		NO_READ_PERMISSION_PATH = "Unable to read the certificate file path {path}."
		NO_FILE_FOUND = "Unable to find the (which certificate) certificate file path {path}." ->(please mention the certificate name)
		NO_VALID_PATHS = "No valid absolute path found for the certificates."
		CHANGES_FOUND = 'Changes found to be applied.'
		FAILED_IMPORT = "Failed to import the certificate file path."
		NO_IMPORT_SUCCESS = "The Secure Boot certificate import operation has failed."

Reply:
1. There's only import supported currently.
2. Will be showing warning if path is not absolute, and he's aware about passing for which parameter, also currently it's not possible, need lots of code change.
3. same apply for no file found.
4. I'll add certificate parameter name when import is not successful.